extensions/blocks/rating-star/rating-meta.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Utilities for the rating block.
+ *
+ * @package Jetpack
+ */
+
+/**
+ * Returns the low fidelity symbol for the block.
+ *
+ * @return string
+ */
+function jetpack_rating_meta_get_symbol_low_fidelity() {
+	return '⭐';
+}
+
+/**
+ * Returns the high fidelity symbol for the block.
+ *
+ * @param array   $attributes Array containing the business hours block attributes.
+ * @param integer $pos Value to render whole and half symbols.
+ * @return string
+ */
+function jetpack_rating_meta_get_symbol_high_fidelity( $attributes, $pos ) {
+	$classname_whole = ( $attributes['rating'] >= ( $pos - 0.5 ) ) ? '' : 'is-rating-unfilled';
+	$classname_half  = ( $attributes['rating'] >= $pos ) ? '' : 'is-rating-unfilled';
+	$color           = empty( $attributes['color'] ) ? 'currentColor' : esc_attr( $attributes['color'] );
+
+	return <<<ELO
+<span>
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
+	<path class="{$classname_whole}" fill="{$color}" stroke="{$color}" d="M12,17.3l6.2,3.7l-1.6-7L22,9.2l-7.2-0.6L12,2L9.2,8.6L2,9.2L7.5,14l-1.6,7L12,17.3z" />
+</svg>
+</span>
+<span>
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
+	<path class="{$classname_half}" fill="{$color}" stroke="{$color}" d="M12,17.3l6.2,3.7l-1.6-7L22,9.2l-7.2-0.6L12,2L9.2,8.6L2,9.2L7.5,14l-1.6,7L12,17.3z" />
+</svg>
+</span>
+ELO;
+
+}
+
+/**
+ * Returns the symbol for the block.
+ *
+ * @param array $attributes Array containing the business hours block attributes.
+ *
+ * @return string
+ */
+function jetpack_rating_meta_get_symbols( $attributes ) {
+	// Output SVGs for high fidelity contexts, then color them according to rating.
+	// These are hidden by default, then unhid when CSS loads.
+	$symbols_hifi = array();
+	for ( $pos = 1; $pos <= $attributes['maxRating']; $pos++ ) {
+		$symbols_hifi[] = '<span style="display: none;">' . jetpack_rating_meta_get_symbol_high_fidelity( $attributes, $pos ) . '</span>';
+	}
+
+	// Output fallback symbols for low fidelity contexts, like AMP,
+	// where CSS is not loaded so the high-fidelity symbols won't be rendered.
+	$symbols_lofi = '';
+	for ( $i = 0; $i < $attributes['rating']; $i++ ) {
+		$symbols_lofi .= jetpack_rating_meta_get_symbol_low_fidelity();
+	}
+
+	return '<p>' . $symbols_lofi . '</p>' . implode( $symbols_hifi );
+}
+
+/**
+ * Dynamic rendering of the block.
+ *
+ * @param array $attributes Array containing the business hours block attributes.
+ *
+ * @return string
+ */
+function jetpack_rating_meta_render_block( $attributes ) {
+	$classname = empty( $attributes['className'] ) ? '' : ' ' . $attributes['className'];
```

Comment:
In `rating-meta` I purposely didn't use anything Jetpack as to be able to copy it verbatim to where [we bundle](https://github.com/Automattic/wpcom-blocks/pull/128) priciness and spiciness -- which are going to be deprecated.

I'm fine with this file changing overtime, but just for this release I want them both equals, as to be able to potentially retrofit any bugfix that migh pop-up, etc.

Reply:
Ok, that makes sense. Do you think you could create a master issue with the things we'll need to update, or one issue for each thing we want to update? 

This way we'll be able to keep track of the things we want to change.

Reply:
Happy to, but not sure which are the things we want to change? This code works fine and takes into account all cases, so using `Jetpack_Gutenberg::block_classes` would be unnecessary.

Reply:
I created #14127 to track this.

While the current implementation works fine, I'd like for things to remain consistent between all our blocks. It also makes it easier to update things later if we ever have to because of a Core update, or decide to change the way we handle classes for all our blocks.